qiskit/quantum_info/synthesis/ion_decompose.py

```diff
@@ -21,8 +21,10 @@
 from qiskit.circuit.library.standard_gates.ry import RYGate
 from qiskit.circuit.library.standard_gates.rx import RXGate
 from qiskit.circuit.library.standard_gates.rxx import RXXGate
+from qiskit.utils.deprecation import deprecate_func
 
 
+@deprecate_func(since="0.46.0", pending=True)
```

Comment:
If this is removed for 1.0 we have to fully deprecate it here 🙂 
```suggestion
@deprecate_func(since="0.46.0")
```

Reply:
OK, so I will replace all pending deprecation warning by deprecation warnings in this PR, since they will be removed in 1.0 (see https://github.com/Qiskit/qiskit/pull/11571)